Update FSC to 29.0

Krzysztof-Cieslak opened the pull request.

Krzysztof-Cieslak committed:
Update FSC to 29.0

Krzysztof-Cieslak committed:
Fix breaking changes

forki commented:
did you consider to add redirects?

enricosada commented:
@forki we need anyway to upgrade FSharpLint, because FCS api changed, so will break at runtime

forki commented:
Makes sense.

Enrico Sada <notifications@github.com> schrieb am Mi., 29. Mai 2019, 17:36:

> @forki <https://github.com/forki> we need anyway to upgrade FSharpLint,
> because FCS api changed, so will break at runtime
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fsharp/FsAutoComplete/pull/386?email_source=notifications&email_token=AAAOANAXA45LRTFYR7NIQ3DPX2PHNA5CNFSM4HP2PHN2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWPXNZI#issuecomment-496989925>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAOANHUKB2NKSTF5OGWXA3PX2PHNANCNFSM4HP2PHNQ>
> .
>

enricosada committed:
update FSharpLint.Core to v0.12.0

forki commented:
Still red. ;-(

enricosada commented:
@forki i know, i am working on that.

the FSharpLint added a breaking change (switcher from xml configs to json configs) we need to address first ( ref https://github.com/fsprojects/FSharpLint/issues/347 )

enricosada committed:
f bump

enricosada committed:
support fsharplint xml configs

enricosada committed:
minimize changes

enricosada committed:
workaround new message format of fsharplint

the message format is `FS01234: info text"` , strip the warning code

enricosada committed:
xmlconfig is read and converted, but hint are not disabled

xmlconfig is read and converted, but hint are not disabled

changed test, based on current fsharplint behaviour

enricosada committed:
the error hint `addTwo2` should be valid suggestion

workaround, remove it from tests.
check if is a FCS regression or is FSAC regression

enricosada committed:
fcs fixed a bug

enricosada committed:
disable Test1Json integration test

it's flaky, the Range sometimes finish at start of newline other times at end of line

like

```
        "Range": {
          "StartColumn": 1,
          "StartLine": 1,
-         "EndColumn": 1,
-         "EndLine": 16
+         "EndColumn": 6,
+         "EndLine": 15
        },
```

Revert "fcs fixed a bug"

This reverts commit b828456109871e1889161d78d503bf48e31f2286.

Krzysztof-Cieslak commented:
GREEN

forki commented:
Awesome. 🛳 it

enricosada commented:
merged.
i added some notes in 45d0ebf about the test changes because need to check these later if are regressions, i tried to check but no definitive answer

Krzysztof-Cieslak commented:
Lack of misspelling  suggestion is breaking change due to https://github.com/dotnet/fsharp/pull/6063

forki commented:
Should still be able to activate, right?

Krzysztof Cieślak <notifications@github.com> schrieb am Mi., 5. Juni 2019,
12:04:

> Lack of misspelling suggestion is breaking change due to
> dotnet/fsharp#6063 <https://github.com/dotnet/fsharp/pull/6063>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fsharp/FsAutoComplete/pull/386?email_source=notifications&email_token=AAAOANFU6DGRF4EF4JD6ZL3PY6FRNA5CNFSM4HP2PHN2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODW7HNDI#issuecomment-499021453>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAOANHYLAZOVHCNKJ5QTODPY6FRNANCNFSM4HP2PHNQ>
> .
>

Krzysztof-Cieslak commented:
I think we may do it like VS - error message without suggestion, calculate suggestion when we calculate quick fix.

forki commented:
Sounds good

Krzysztof Cieślak <notifications@github.com> schrieb am Mi., 5. Juni 2019,
12:57:

> I think we may do it like VS - error message without suggestion, calculate
> suggestion when we calculate quick fix.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fsharp/FsAutoComplete/pull/386?email_source=notifications&email_token=AAAOANH4P3OTAFXXRIHUA6DPY6LZNA5CNFSM4HP2PHN2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODW7LG4Q#issuecomment-499037042>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAOANDP2YUWVBNPIDEW2WDPY6LZNANCNFSM4HP2PHNQ>
> .
>

enricosada commented:
Really sad this feature is gone. It was really really nice as is.

I like to have the solution in the tooltip directly, in the quickfix is too much work for me as user, while in tooltip helps while typing

Also sometimes the suggestion is not really the fix, just a false positive, so dunno, i like quickfix because these always works.

Krzysztof-Cieslak commented:
Cc: @cartermp
